### PR TITLE
Charge new cards immediately if necessary

### PR DESCRIPTION
**Pull request by leogrim**
@andrewconner @cvializ

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/PaymentsController.scala [hunk 1]
```diff
@@ -56,11 +58,21 @@ class PaymentsController @Inject() (
...
+                val futureCharge = if (lastPaymentFailed) paymentCommander.processAccount(request.orgId).imap { case (_, event) => event.paymentCharge } else Future.successful(None)
+                futureCharge.imap { charge =>
+                  Ok(Json.obj(
+                    "card" -> cardInfo,
+                    "charge" -> charge
+                  ))
+                }
+              case Failure(InvalidChange(msg)) => Future.successful(BadRequest(Json.obj("error" -> msg)))
+              case Failure(ex) => Future.failed(ex)
+            }
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/PaymentsController.scala:76]: @cvializ updated this API, let me know if it makes sense and whether I can merge this

Comment (cvializ) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/PaymentsController.scala:76]: The charge field next to the card field makes sense, but I'm having a hard time figuring out the structure of the charge value. Could you give me a sample value for the `"charge"`?

Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/PaymentsController.scala:76]: Whoops, sorry, that's just a dollar amount, either `{cents: JsNumber}` or `null` if there wasn't a charge

Comment (cvializ) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/PaymentsController.scala:76]: ah, got it! lgtm :+1:

**Pull request merged**